File: javascript/src/model/RegionOfInterest.js
// require files in Node.js environment

if (typeof module === 'object' && module.exports) {
  
}




//export module
if ( typeof define === "function" && define.amd ) {
  define('RegionOfInterest', ['jquery'],
    function($) {
      return RegionOfInterest;
   });
}


var RegionOfInterest = function RegionOfInterest() { 
  var self = this;
  
  /**
   * datatype: Integer
   **/
  self.x = null;
  
  /**
   * datatype: Integer
   **/
  self.y = null;
  
  /**
   * datatype: Integer
   **/
  self.width = null;
  
  /**
   * datatype: Integer
   **/
  self.height = null;
  
  
  self.constructFromObject = function(data) {
    if (!data) {
      return;
    }
    
    self.x = data.x;
    
    self.y = data.y;
    
    self.width = data.width;
    
    self.height = data.height;
    
  }

  
  /**
   * @return {Integer}
   **/
  self.getX = function() {
    return self.x;
  }

  /**
   * @param {Integer} x
   **/
  self.setX = function (x) {
    self.x = x;
  }
  
  /**
   * @return {Integer}
   **/
  self.getY = function() {
    return self.y;
  }

  /**
   * @param {Integer} y
   **/
  self.setY = function (y) {
    self.y = y;
  }
  
  /**
   * @return {Integer}
   **/
  self.getWidth = function() {
    return self.width;
  }

  /**
   * @param {Integer} width
   **/
  self.setWidth = function (width) {
    self.width = width;
  }
  
  /**
   * @return {Integer}
   **/
  self.getHeight = function() {
    return self.height;
  }

  /**
   * @param {Integer} height
   **/
  self.setHeight = function (height) {
    self.height = height;
  }
  

  self.toJson = function () {
    return JSON.stringify(self);
  }
}

if (typeof module === 'object' && module.exports) {
  module.exports = RegionOfInterest;
}

File: java/src/main/java/com/openalpr/api/models/RegionOfInterest.java
package com.openalpr.api.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;





@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-02-17T14:18:32.383-05:00")
public class RegionOfInterest   {
  
  private Integer x = null;
  private Integer y = null;
  private Integer width = null;
  private Integer height = null;

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("x")
  public Integer getX() {
    return x;
  }
  public void setX(Integer x) {
    this.x = x;
  }

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("y")
  public Integer getY() {
    return y;
  }
  public void setY(Integer y) {
    this.y = y;
  }

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("width")
  public Integer getWidth() {
    return width;
  }
  public void setWidth(Integer width) {
    this.width = width;
  }

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("height")
  public Integer getHeight() {
    return height;
  }
  public void setHeight(Integer height) {
    this.height = height;
  }

  

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RegionOfInterest regionOfInterest = (RegionOfInterest) o;
    return Objects.equals(x, regionOfInterest.x) &&
        Objects.equals(y, regionOfInterest.y) &&
        Objects.equals(width, regionOfInterest.width) &&
        Objects.equals(height, regionOfInterest.height);
  }

  @Override
  public int hashCode() {
    return Objects.hash(x, y, width, height);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class RegionOfInterest {\n");
    
    sb.append("    x: ").append(toIndentedString(x)).append("\n");
    sb.append("    y: ").append(toIndentedString(y)).append("\n");
    sb.append("    width: ").append(toIndentedString(width)).append("\n");
    sb.append("    height: ").append(toIndentedString(height)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

